### PR TITLE
Update user by calling API

### PR DESCRIPTION
**Pull request by jinety**

### CODE REVIEW

File: big-practice/src/js/dashboard.js [hunk 1]
```diff
@@ -125,6 +143,7 @@ const renderTable = () => {
       updateButtons.forEach((item) => {
         item.addEventListener('click', () => {
           showModal();
+          getMovie(item);
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/dashboard.js]: Do you think hide/showElement should be after we load getMovie?

File: big-practice/src/js/dashboard.js [hunk 2]
```diff
@@ -182,6 +223,36 @@ const handleCreateForm = () => {
...
+  const url = `${MoviesApi}?name=${name}`;
+
+  if (!isValidForm()) {
+    return;
+  }
+
+  fetch(url, { method: 'GET' })
+    .then((response) => response.json())
+    .then((movieList) => {
+      if (movieList.length > 1) {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/dashboard.js]: Why don't use else if?

File: big-practice/src/js/dashboard.js [hunk 3]
```diff
@@ -182,6 +223,36 @@ const handleCreateForm = () => {
     });
 };
 
+const handleUpdateForm = () => {
+  const customMovieId = form.getAttribute('data-id');
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/dashboard.js]: customMovieId is not good name, it should not be CUSTOM

File: big-practice/src/js/dashboard.js [hunk 4]
```diff
@@ -182,6 +223,36 @@ const handleCreateForm = () => {
     });
 };
 
+const handleUpdateForm = () => {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/dashboard.js:226]: Is this handle for empty data as creating movie?

File: big-practice/src/js/dashboard.js [hunk 5]
```diff
@@ -93,6 +94,23 @@ const isValidForm = () => {
   return isValid;
 };
 
+const getMovie = (item) => {
```
Comment (hieungo-agilityio) [inline big-practice/src/js/dashboard.js]: The param should be id of movie, no need to pass item object.

Comment (hieungo-agilityio) [reply big-practice/src/js/dashboard.js]: Add comment for func please.

File: big-practice/src/js/dashboard.js [hunk 6]
```diff
@@ -199,6 +270,10 @@ cancelBtn.addEventListener('click', () => {
   hideModal();
 });
 
+updateBtn.addEventListener('click', () => {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/dashboard.js:214]: Missing comment

File: big-practice/src/js/dashboard.js (hunk 4 above)
Comment (phuocduong-agilityio) [inline big-practice/src/js/dashboard.js:175]: Where is comment? Pls check around

File: big-practice/src/js/dashboard.js [hunk 7]
```diff
@@ -155,6 +174,28 @@ const createMovie = (data) => {
     .catch((error) => alert('An error occurred while creating movie', error));
 };
 
+const updateForm = () => {
```
Comment (hieungo-agilityio) [inline big-practice/src/js/dashboard.js]: What is the purpose of `updateForm`? The func name doesn't make sense.

File: big-practice/src/js/dashboard.js (hunk 4 above)
Comment (jinety) [reply big-practice/src/js/dashboard.js:175]: fixed

File: big-practice/src/js/dashboard.js (hunk 6 above)
Comment (jinety) [reply big-practice/src/js/dashboard.js:214]: fixed

File: big-practice/db.json [hunk 8]
```diff
@@ -21,4 +21,4 @@
       "id": 2
     }
   ]
-}
+}
```
Comment (hieungo-agilityio) [inline big-practice/db.json]: add empty line

File: big-practice/src/js/dashboard.js [hunk 9]
```diff
@@ -182,6 +225,33 @@ const handleCreateForm = () => {
...
+  const formMovieId = form.getAttribute('data-id');
+
+  if (!isValidForm()) {
+    return;
+  }
+
+  fetch(url, { method: 'GET' })
+    .then((response) => response.json())
+    .then((movieList) => {
+      if (movieList.length === 0 || movieList[0].id === parseFloat(formMovieId)) {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/dashboard.js]: Should use parseInt, not float

**Pull request merged**